src/functions/weekly-check/handler.ts

```diff
@@ -0,0 +1,126 @@
+import fetch from "node-fetch";
+import { DateTime } from "luxon";
+import { OnCallEntry } from "../../types"
+import S3Utils from "@libs/s3-utils";
+import { middyfy } from "@libs/lambda";
+import Config from "src/app/config";
+
+/**
+ * Schedule from Splunk
+ */
+interface Schedule {
+    team: {
+        name: string;
+        slug: string;
+    },
+    schedules: [
+        {
+            policy: {
+                name: string;
+                slug: string;
+            },
+            schedule: [
+                {
+                    onCallUser: {
+                        username: string;
+                    },
+                    overrideOnCallUser?: {
+                        username: string;
+                    },
+                    onCallType: string;
+                    rotationName: string;
+                    shiftName: string;
+                    shiftRoll: string;
+                    rolls: [
+                        {
+                            start: string;
+                            end: string;
+                            onCallUser: {
+                                username: string;
+                            },
+                            isRoll: boolean
+                        }
+                    ]
+                }
+            ],
+            overrides: any[]
+        }
+    ]
+};
+
+/**
+ * Resolve next week from schedule
+ * 
+ * @param schedule schedule
+ * @param nextThursday next Thursday
+ * @param policyName name of policy
+ * @returns week from schedule
+ */
+const getNextWeekFromSchedule = (schedule: Schedule, nextThursday: DateTime, policyName: string) => {
+    const scheduleSchedule = schedule.schedules.find(shedule => shedule.policy.name === policyName)?.schedule[0];
+    if (!scheduleSchedule) {
+        return null;
+    }
+
+    const onCallUser = scheduleSchedule.onCallUser;
+    const overrideOnCallUser = scheduleSchedule.overrideOnCallUser;
+    const weekNumber = nextThursday.weekNumber;
+    
+    return {
+        week: weekNumber,
+        user: overrideOnCallUser == null ? onCallUser.username : overrideOnCallUser.username
+    };
+};
+
+/**
+ * Lambda for checking who is on call this week
+ * 
+ * @param event event
+ */
+export const weeklyCheckHandler = async () => {
+    const { SPLUNK_SCHEDULE_POLICY_NAME, SPLUNK_TEAM_ONCALL_URL, SPLUNK_API_ID, SPLUNK_API_KEY } = process.env;
+    if (!SPLUNK_SCHEDULE_POLICY_NAME || !SPLUNK_TEAM_ONCALL_URL || !SPLUNK_API_ID || !SPLUNK_API_KEY) {
+        throw new Error('Missing environment variables');
+    }
+
+    const splunkTeamOnCallUrl = SPLUNK_TEAM_ONCALL_URL;
+    
+    const schedule = await (await fetch(`${splunkTeamOnCallUrl}/schedule?daysForward=4&daysSkip=3`, {
+        headers: {
+            'X-VO-Api-Id': SPLUNK_API_ID,
+            'X-VO-Api-Key': SPLUNK_API_KEY,
+            'Accept': 'application/json'
+        } 
+    })).json() as Schedule;
+
+    const nextThursday = DateTime.now().plus({ days: 4 });
```

Comment:
I couldn't think much better of a name, I thought about "fourDaysFromSunday" but this could sound as cryptic. I opted to keep this variable name and update its description in getNextWeekFromSchedule JSDOC, where it is used as a parameter.